draft-packages/modal/KaizenDraft/Modal/Primitives/GenericModal.tsx

```diff
@@ -289,4 +289,28 @@ const createAriaHider = (dialogNode: HTMLElement): (() => void) => {
   }
 }
 
-export default GenericModalContainer
+const ClientOnly = ({
+  children,
+}: {
+  readonly children: React.ReactNode
+}): JSX.Element => {
+  const [hasMounted, setHasMounted] = React.useState(false)
+
+  React.useEffect(() => {
```

Comment:
Can we please avoid using side effects to deal with server-side rendering and client-side rendering hydration errors?

Next.js recommends using https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr. This simple solution can be used on any component. 

If there are reusability concerns, can we please create a util function like this

```
export const isSSR = () => typeof window === "undefined"
```

and use it directly without using `useEffect`

```
return <>{!isSSR && children}</>
```

Reply:
Oh I forgot about dynamic imports. That's all that's needed for now. I'll close this PR